projects/vdk-plugins/vdk-postgres/tests/conftest.py

```diff
@@ -44,17 +42,34 @@ def docker_compose_file(pytestconfig):
         VDK_POSTGRES_PORT: "5432",
     },
 )
-def postgres_service(docker_ip, docker_services):
+def postgres_service(request):
     """Ensure that Postgres service is up and responsive."""
-    runner = CliEntryBasedTestRunner(postgres_plugin)
+    port = int(os.environ[VDK_POSTGRES_PORT])
+    container = (
+        DockerContainer("postgres:latest")
+        .with_bind_ports(port, port)
+        .with_env("POSTGRES_PASSWORD", os.environ[VDK_POSTGRES_PASSWORD])
+    )
 
-    # give the server some time to start before checking if it is ready
-    # before adding this sleep there were intermittent fails of the CI/CD with error:
-    # requests.exceptions.ConnectionError:
-    #   ('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))
-    # More info: https://stackoverflow.com/questions/383738/104-connection-reset-by-peer-socket-error-or-when-does-closing-a-socket-resu
-    time.sleep(3)
+    try:
+        container.start()
+        runner = CliEntryBasedTestRunner(postgres_plugin)
+        wait_for(partial(_is_responsive, runner))
+        # wait 2 seconds to make sure the service is up and responsive
+        # might be unnecessary but it's out of abundance of caution
+        time.sleep(2)
+        print(
+            f"Postgres service started on port {container.get_exposed_port(port)} and host {container.get_container_host_ip()}"
+        )
+    except Exception as e:
+        print(f"Failed to start Postgres service: {e}")
```

Comment:
I don't like the way you are logging the exception and also throwing it. 
This always results in lots of noise in our code.

Reply:
I think its best to create a new exception which includes the logs. 
I think its best to remove  `print(f"Failed to start Postgres service: {e}")`

Reply:
Make sense.